judge/comments.py

```diff
@@ -65,6 +65,10 @@ def post(self, request, *args, **kwargs):
         self.object = self.get_object()
         page = self.get_comment_page()
 
+        if (CommentLock.objects.filter(page=self.get_comment_page()).exists() and
+            not (self.request.user.is_superuser or self.request.user.has_perm('judge.override_comment_lock'))):
```

Comment:
Should probably move the condition to a method called `is_comment_locked` or something similar, since you duplicate it again in `get_context_data`.